main.go

```diff
@@ -166,26 +173,21 @@ func main() {
 		})
 	}
 
+	tlsConfig := opts.TLSConfig
+
 	telemetryMux := buildTelemetryServer(ksmMetricsRegistry)
-	telemetryServer := http.Server{Handler: telemetryMux}
 	telemetryListenAddress := net.JoinHostPort(opts.TelemetryHost, strconv.Itoa(opts.TelemetryPort))
-	telemetryLn, err := net.Listen("tcp", telemetryListenAddress)
-	if err != nil {
-		klog.Fatalf("Failed to create Telemetry Listener: %v", err)
-	}
+	telemetryServer := http.Server{Handler: telemetryMux, Addr: telemetryListenAddress}
+
 	metricsMux := buildMetricsServer(m, durationVec)
-	metricsServer := http.Server{Handler: metricsMux}
 	metricsServerListenAddress := net.JoinHostPort(opts.Host, strconv.Itoa(opts.Port))
-	metricsServerLn, err := net.Listen("tcp", metricsServerListenAddress)
-	if err != nil {
-		klog.Fatalf("Failed to create MetricsServer Listener: %v", err)
-	}
+	metricsServer := http.Server{Handler: metricsMux, Addr: metricsServerListenAddress}
 
 	// Run Telemetry server
 	{
 		g.Add(func() error {
 			klog.Infof("Starting kube-state-metrics self metrics server: %s", telemetryListenAddress)
-			return telemetryServer.Serve(telemetryLn)
+			return web.ListenAndServe(&telemetryServer, tlsConfig, promLogger{})
```

Comment:
Last comment. Let's use one instance of promLogger and assign that to a variable. We can reuse it across these methods

Reply:
Done!